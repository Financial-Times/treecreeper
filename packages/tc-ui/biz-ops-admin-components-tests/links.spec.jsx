// const schema = require('@financial-times/tc-schema-sdk');
// const React = require('react');
// require('../../../src/lib/configure-schema');

// const {
// 	BizOpsLink,
// 	BizOpsLinks,
// } = require('../../../src/templates/components/primitive-components.jsx');

// const { render } = require('../../testHelpers/component');

// describe('Biz Ops Links', () => {
// 	beforeAll(() => schema.refresh());
// 	describe('<BizOpsLink />', () => {
// 		it('Display a fully specified link', () => {
// 			const component = render(
// 				<BizOpsLink
// 					type="System"
// 					value={{ code: 'a-code', name: 'a-name' }}
// 				/>,
// 			);
// 			expect(component).toMatchSnapshot();
// 		});

// 		it('Display a link with no name', () => {
// 			const component = render(
// 				<BizOpsLink type="System" value={{ code: 'a-code' }} />,
// 			);
// 			expect(component).toMatchSnapshot();
// 		});

// 		it('Display a link with no code', () => {
// 			const component = render(
// 				<BizOpsLink type="System" value={{ name: 'a-name' }} />,
// 			);
// 			expect(component).toMatchSnapshot();
// 		});

// 		it('Display a link with no code or name', () => {
// 			const component = render(<BizOpsLink type="System" />);
// 			expect(component).toMatchSnapshot();
// 		});
// 	});

// 	const baseObject = () => ({
// 		list01: [
// 			{ code: 'code01', name: 'name01' },
// 			{ code: 'code02', name: 'name02' },
// 			{ code: 'code03', name: 'name03' },
// 			{ code: 'code04', name: 'name04' },
// 			{ code: 'code05', name: 'name05' },
// 		],
// 	});

// 	describe('<BizOpsLinks />', () => {
// 		it('Display a fully specified list', () => {
// 			const props = baseObject();
// 			const component = render(
// 				<BizOpsLinks
// 					type="System"
// 					label="a-label"
// 					value={props.list01}
// 				/>,
// 			);
// 			expect(component).toMatchSnapshot();
// 		});

// 		it('Display a link with no list', () => {
// 			const component = render(
// 				<BizOpsLinks type="System" label="a-label" />,
// 			);
// 			expect(component).toMatchSnapshot();
// 		});

// 		it('Display a link with no label', () => {
// 			const props = baseObject();
// 			const component = render(
// 				<BizOpsLinks type="System" value={props.list01} />,
// 			);
// 			expect(component).toMatchSnapshot();
// 		});

// 		it('Display a link with no list content', () => {
// 			const props = {};
// 			const component = render(
// 				<BizOpsLinks type="System" value={props} />,
// 			);
// 			expect(component).toMatchSnapshot();
// 		});
// 	});
// });
