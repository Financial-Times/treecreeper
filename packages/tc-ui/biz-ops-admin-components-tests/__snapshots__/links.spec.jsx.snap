// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Biz Ops Links <BizOpsLink /> Display a fully specified link 1`] = `
<a href="/System/a-code">a-name</a>
<span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown service tier</span>
<span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>
`;

exports[`Biz Ops Links <BizOpsLink /> Display a link with no code 1`] = `"Error: unable to construct link"`;

exports[`Biz Ops Links <BizOpsLink /> Display a link with no code or name 1`] = `"Error: unable to construct link"`;

exports[`Biz Ops Links <BizOpsLink /> Display a link with no name 1`] = `
<a href="/System/a-code">a-code</a>
<span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown service tier</span>
<span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>
`;

exports[`Biz Ops Links <BizOpsLinks /> Display a fully specified list 1`] = `
<ul class="o-layout__unstyled-element biz-ops-links">
  <li class="active">
    <a href="/System/code01">name01</a>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown service tier</span>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>
  </li>
  <li class="active">
    <a href="/System/code02">name02</a>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown service tier</span>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>
  </li>
  <li class="active">
    <a href="/System/code03">name03</a>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown service tier</span>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>
  </li>
  <li class="active">
    <a href="/System/code04">name04</a>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown service tier</span>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>
  </li>
  <li class="active">
    <a href="/System/code05">name05</a>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown service tier</span>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>
  </li>
</ul>
`;

exports[`Biz Ops Links <BizOpsLinks /> Display a link with no label 1`] = `
<ul class="o-layout__unstyled-element biz-ops-links">
  <li class="active">
    <a href="/System/code01">name01</a>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown service tier</span>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>
  </li>
  <li class="active">
    <a href="/System/code02">name02</a>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown service tier</span>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>
  </li>
  <li class="active">
    <a href="/System/code03">name03</a>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown service tier</span>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>
  </li>
  <li class="active">
    <a href="/System/code04">name04</a>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown service tier</span>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>
  </li>
  <li class="active">
    <a href="/System/code05">name05</a>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown service tier</span>
    <span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>
  </li>
</ul>
`;

exports[`Biz Ops Links <BizOpsLinks /> Display a link with no list 1`] = `"Error: unable to construct links"`;

exports[`Biz Ops Links <BizOpsLinks /> Display a link with no list content 1`] = `"Error: unable to construct links"`;
